src/paymentservice/charge.js

```diff
@@ -1,24 +1,39 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
-const {context, propagation, trace, metrics} = require('@opentelemetry/api');
+const { context, propagation, trace, metrics } = require('@opentelemetry/api');
 const cardValidator = require('simple-card-validator');
 const { v4: uuidv4 } = require('uuid');
 
 const { OpenFeature } = require('@openfeature/server-sdk');
-const { FlagdProvider} = require('@openfeature/flagd-provider');
+const { FlagdProvider } = require('@openfeature/flagd-provider');
 const flagProvider = new FlagdProvider();
 
 const logger = require('./logger');
 const tracer = trace.getTracer('paymentservice');
 const meter = metrics.getMeter('paymentservice');
-const transactionsCounter = meter.createCounter('app.payment.transactions')
+const transactionsCounter = meter.createCounter('app.payment.transactions');
+
+const TENANT_LEVEL = ['gold', 'silver', 'bronze'];
+
+/** Return random element from given array */
+function random(arr) {
+  const index = Math.floor(Math.random() * arr.length);
+  return arr[index];
+}
```

Comment:
Ideally we would configure this directly on the Feature Flag, using fractional evaluation: https://openfeature.dev/specification/glossary/#fractional-evaluation

Reply:
Excellent idea, will look into this, thank you :-)

Reply:
Or how about doing something like this:

<img width="518" alt="image" src="https://github.com/user-attachments/assets/88159953-a4b6-408e-adb7-d5cfd57c1103">

Using the following `flagd` configuration:
```
    "paymentServiceFailure": {
      "description": "Fail payment service charge requests n%",
      "state": "ENABLED",
      "variants": {
        "100%": 1.00,
        "90%": 0.95,
        "75%": 0.75,
        "50%": 0.50,
        "25%": 0.25,
        "10%": 0.10,
        "off": 0.00
      },
      "defaultVariant": "off"
    },
```